File: 20230929_Friday/20230929.go
// Leetcode: 896. Monotonic Array.
// https://leetcode.com/problems/monotonic-array
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 140 ms, faster than 55.32% of Go online submissions for Monotonic
// Array.
// Memory Usage: 8.5 MB, less than 89.36% of Go online submissions for Monotonic
// Array.
// 2023.09.29 Daily Challenge.
package main

func isMonotonic(nums []int) bool {
	x := nums
	up, down := false, false
	n := len(x)
	for i := 1; i < n; i += 1 {
		if up && down {
			return false
		}
		if x[i-1] > x[i] {
			down = true
		}
		if x[i-1] < x[i] {
			up = true
		}
	}
	if up && down {
		return false
	}
	return true
}

File: 20230929_Friday/20230929.rb
# Leetcode: 896. Monotonic Array.
# https://leetcode.com/problems/monotonic-array
# = = = = = = = = = = = = = =
# Accpeted.
# Thanks God, Jesus Christ!
# = = = = = = = = = = = = = =
# Runtime: 151 ms, faster than 35.48% of Ruby online submissions for Monotonic
# Array.
# Memory Usage: 223.4 MB, less than 29.03% of Ruby online submissions for
# Monotonic Array.
# 2023.09.29 Daily Challenge.
# @param {Integer[]} nums
# @return {Boolean}
def is_monotonic(nums)
  x = nums
  up = false
  down = false
  (1...x.size).each do |i|
    return false if up && down
    up = true if x[i-1] < x[i]
    down = true if x[i-1] > x[i]
  end
  return false if up && down
  return true
end
